### ISSUE TITLE
Group messages unreliable when phone is offline for a while

### ISSUE THREAD

**Issue by Wasseranomalie** [signalapp/Signal-Android#3312]
I am in a group of 3 people. If the other 2 are discussing something while my phone isn´t connected to a network and i go online after a while, i get the messages, but some messages are missing (and the messages are in the wrong order, but i think an issue regarding this is already there).

I don´t know if the folliwing is related to this issue, but when i lock TextSecure, i get a notification about new messages (but there are no new messages visible to me).

![notification](https://cloud.githubusercontent.com/assets/6809794/7939045/3b115dd6-0949-11e5-9542-45a33fec4028.png)

This issue happend already several times, so maybe because of that there are new messages in two conversations ("Unterhaltungen").

Samsung Galaxy S2, OmniROM Android 4.4.4, TextSecure 2.14.5

**Comment by rhodey** [comment]
hey @Wasseranomalie, the new message notifications are probably a separate issue but I'm interested in hearing more about that because we believed we had solved that problem for good.

You said you've had this problem with new message notifications before? Can you remember what specifically made them go away, an update or something? Also, did these new message notifications show up after you're phone came back online and received a bunch of group messages?

**Comment by Wasseranomalie** [comment]
Sorry for this misunderstanding, but i meant, that i had this group-messages-issue several times. I have this notification-bug since a while now (i unfortunately don´t know anymore when it first appeared. It never disappeared until now). I linked this two bugs, because i think that i first saw this notification-bug after locking TextSecure because i hoped, that this might solve the problem of the missing messages (after i got a bunch of them because my phone was offline). But i don´t know if that issue was the cause of this notification-bug.

**Comment by dev-tee** [comment]
It seems like this issue isn't limited to group chats. I've had similar problems since wednesday.
Today I've tried to send a few random messages to my phone while it was in airplane mode. Once I turned off airplane mode and received all the messages, I opened the app and read them. After locking the app again, a notification showed up telling me that there are 5 new messages. Checking my messages again, I found that there are no actual new messages.
Hitting "Mark everything as read" doesn't seem to solve this and letting the app lock itself after my password times out also appears to have no effect.
As a sidenote, the number of alleged new messages and the messages I received during my test is significantly lower. About 60 received vs 5 "new" messages.

Device info: Sony Ericsson Xperia NEO, LegacyXperia CM11 Android 4.4.4, TextSecure 2.16.4
